core/src/main/java/io/questdb/griffin/engine/functions/eq/EqShortFunctionFactory.java

```diff
@@ -61,7 +61,7 @@ public Func(Function left, Function right) {
 
         @Override
         public boolean getBool(Record rec) {
-            return negated != (left.getShort(rec) == right.getShort(rec));
+            return !right.isNullConstant() && (negated != (left.getShort(rec) == right.getShort(rec)));
```

Comment:
null constant check should be done in the Factory instead.  When either `left` or `right` is null, the predicate becomes constant `false` and doesn't have to be computed on every row. It should also become a `constant` so that SQL optimiser is allowed to make optimisation around it.

Here is an example: https://github.com/questdb/questdb/blob/master/core/src/main/java/io/questdb/griffin/engine/functions/regex/AbstractLikeStrFunctionFactory.java#L88

Reply:
Ah ok , let me make those changes

Reply:
I've made the suggested changes @bluestreak01